src/inputs/Button/Button.tsx

```diff
@@ -26,18 +31,63 @@ export const Button = ({
     }
   };
 
+  const getIsLoadingClassForSpecificVariant = (): string => {
+    if (isLoading) {
+      return variant === "primary"
+        ? styles.isLoadingPrimary
+        : styles.isLoadingSecondary;
+    }
+  };
+
   const dynamicClasses = [
     styles.template,
     variantClass,
     getDisabledClassForSpecificVariant(),
+    getIsLoadingClassForSpecificVariant(),
   ].join(" ");
 
+  // cons for icon
+
+  const getIconClassForSpecificVariant = ():string => {
+    return icon === true
+      ? styles.iconShow
+      : styles.iconHide
+  };
+
+  const getIfIsSpinner = ():string => {
+    return isLoading && !disabled === true
```

Comment:
**suggestion**: you can simplify this to: `return isLoading && !disabled ? styles.iconSpinner : styles.iconPlus`